File: code/stable/GOOLTest/java/PatternTest/PatternTest/Observer.java
package PatternTest;

/** \brief This is an arbitrary class acting as an Observer
*/
public class Observer {
    public int x;
    
    public Observer() {
        this.x = 5;
    }
    
    public void printNum() {
        System.out.println(this.x);
    }
    
    public int getX() {
        return this.x;
    }
    
    public void setX(int x) {
        this.x = x;
    }
}

File: code/stable/gooltest/cpp/PatternTest/Observer.cpp
#include "Observer.hpp"

#include <iostream>

Observer::Observer() : x(5) {
}

void Observer::printNum() {
    std::cout << this->x << std::endl;
}

int Observer::getX() {
    return this->x;
}

void Observer::setX(int x) {
    this->x = x;
}
